template.go

```diff
@@ -10,21 +10,23 @@ import (
 
 // {{ .Name }}Mock is a mock implementation of the {{ .Name }}
 // interface.
-type {{ .Name }}Mock struct {
+type {{ .Name }}Mock{{ .TypeParams }} struct {
 	T *testing.T
 	{{- range .Methods }}
 	{{ .Name }}Stub func({{ .Params }}) {{ .Results }}
 	{{ .Name }}Called int32
 	{{- end }}
 }
 
+{{ if not .TypeParams }}
 var _ {{ .Name }} = &{{ .Name }}Mock{}
+{{ end }}
```

Comment:
> Generated mock files currently include assignment of the mock type to a variable of the interface type, as a kind of unit test that the mock actually implements the interface. It's unclear how in general we should generate these checks in the case of generic mocks.

Major caveat: this might be a terrible idea :joy:

So, the idea behind this test we add to the mock files is that we want the compiler to verify that a value of type `MyMock` can be assigned to a variable of type `MyInterface`, thereby proving that the type `MyMock` implements the `MyInterface` interface.

Excuse the abuse of notation, but this is essentially a propositional logic statement:
`MyMock satisfies MyInterface` (If you have a `MyMock`, then you have a `MyInterface`).

When we add a type constraint set `T` to the definition of `MyInterface`, I think the problem is that the desired statement becomes a first order logic expression:
`∀X (X satisfies T) -> MyMock[X] satisfies MyInterface[X]` (For all types X, if X satisfies the type constraints T, then a MyMock[X] will satisfy the concrete interface type MyInterface[X])

I totally agree that deciding what X to pick to create this test isn't super helpful. (For one, actually picking one is hard, and for two, it's hard to say that the selected X would guarantee that the mock would satisfy the interface for all allowed values of X).

However... could we just generate code here that essentially implements the full first order logic expression? :thinking: 

e.g., imagine that given an interface definition `MyInterface[T any, U comparable]`, we generated a snippet like the following:
```go
func _[T any, U comparable]() {
    var _ MyInterface[T, U] = &MyInterfaceMock[T, U]{}
}
```

My feeling is that this wouldn't compile if the mock didn't have the same type constraints as the `MyInterface` type :thinking: I think this would be easy enough to generate, too, since we have `TypeParams.String` and `TypeParams.Names`.

What do you think of this approach?

Reply:
Great idea! 2e78e2bb06f6e584ed97dd37c2d1d4f4e8f2fbc3 I also added a generated comment to explain the purpose of these `_` assignments - let me know what you think.

Reply:
Looks great to me, thanks for throwing that in!